src/services/helpers.ts

```diff
@@ -74,15 +75,22 @@ export const shuffleArray = <T>(array: T[]): T[] => {
   return shuffled
 }
 
-const getDoneExercisesByProgress = (disciplineId: number, progress: Progress): number => {
+const getDoneExercisesNumberInProgressByDiscipline = (disciplineId: number, progress: Progress): number => {
   const progressOfDiscipline = progress[disciplineId]
+  const exerciseKeyForWordChoice = ExerciseKeys.wordChoiceExercise
   return progressOfDiscipline
-    ? Object.keys(progressOfDiscipline).filter(item => progressOfDiscipline[item] !== undefined).length
+    ? Object.keys(progressOfDiscipline).filter(item => {
```

Comment:
Your function here does not really make sense:
1) You are iterating over the Object keys of the progress of on discipline. So you are iterating over the exercise keys: 0, 1, 2, so `item.length` is always undefined.
2) You are comparing the score of a exercise with `ExerciseKeys.wordChoiceExercise`. This does not make sense, as a score is "how good a exericse is done" and the ExerciseKey is "which exercise is done". These two cannot be compared.

What you want to do is: Filter and use only those entries, where either the ExerciseKey is vocabularyList OR the score is bigger than the threshold.

Reply:
@lunars97 Do you want help with implementing the suggested function?